src/Http/Controllers/VoyagerSettingsController.php

```diff
@@ -84,18 +84,24 @@ public function update(Request $request)
         $this->authorize('edit', Voyager::model('Setting'));
 
         $settings = Voyager::model('Setting')->all();
-
+        
         foreach ($settings as $setting) {
+            $field = str_replace('.', '_', $setting->key);
+
+            if ($setting->type == 'image' && request($field) == null) {
+                continue;
+            }
+
             $content = $this->getContentBasedOnType($request, 'settings', (object) [
                 'type'    => $setting->type,
-                'field'   => str_replace('.', '_', $setting->key),
+                'field'   => $field,
                 'details' => $setting->details,
                 'group'   => $setting->group,
             ]);
 
```

Comment:
Move your condition from line 91 here and use
`if ($setting->type == 'image' && $content == null)`
This makes everything else unnecessary.